insights/core/__init__.py

```diff
@@ -98,6 +102,112 @@ def default_parser_deserializer(_type, data):
     return obj
 
 
+def find_main(confs, name):
+    for c in confs:
+        if c.file_name == name:
+            return c
+
+
+def flatten(docs, pred):
+    def inner(children):
+        results = []
+        for c in children:
+            if select(pred)([c]) and c.children:
+                results.extend(inner(c.children))
+            else:
+                results.append(c)
+        return results
+    return inner(docs)
+
+
+class ConfigComponent(object):
+    def select(self, *queries, **kwargs):
```

Comment:
Would you mind to add documentation to these new classes including some information to help developers understand how to use them? Thanks!